libraries/chain/finality/finalizer.cpp

```diff
@@ -99,7 +98,7 @@ bool finalizer::maybe_update_fsi(const block_state_ptr& bsp) {
    auto& final_on_strong_qc_block_ref = bsp->core.get_block_reference(bsp->core.final_on_strong_qc_block_num);
    if (final_on_strong_qc_block_ref.timestamp > fsi.lock.timestamp && bsp->timestamp() > fsi.last_vote.timestamp) {
       fsi.lock = final_on_strong_qc_block_ref;
```

Comment:
I believe we should set the timestamp on `fsi.lock`. Also on line 83.

Reply:
`fsi.lock` is from `final_on_strong_qc_block_ref` which already contains the timestamp. What other timestamp to set?

Reply:
Oh, my bad, sorry!